### PR TITLE
[JS] feat: update teamschef bot sample to use msi for remote

### PR DESCRIPTION
**Pull request by yukun-dong**
## Linked issues
#minor 
 # (issue number)

## Details

Update this sample to use managed identity for remote to avoid creating secret for bot.

#### Change details

> Describe your changes, with screenshots and code snippets as appropriate

**code snippets**:

**screenshots**:

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### Additional information

> Feel free to add other relevant information below

### CODE REVIEW

**Comment by corinagum** [comment]
@yukun-dong, thanks for the info. We're following up internally on our team -- the internal MSFT solution is to have a dev tenant right now, and we don't want to force customers to use msi if security requirements don't make it necessary. Will get back to you.

**Comment by corinagum** [comment]
@yukun-dong, thanks for your patience. I spoke with the team, and if your team is required to make this change, we would prefer if Teams-fx team hosts their own version of chef bot. Therefore, I will close this PR. Thank you!

File: js/samples/04.ai-apps/a.teamsChefBot/src/index.ts [hunk 1]
```diff
@@ -32,9 +32,10 @@ config({ path: ENV_FILE });
 const adapter = new TeamsAdapter(
     {},
     new ConfigurationServiceClientCredentialFactory({
+        MicrosoftAppType: process.env.BOT_TYPE,
         MicrosoftAppId: process.env.BOT_ID,
         MicrosoftAppPassword: process.env.BOT_PASSWORD,
-        MicrosoftAppType: 'MultiTenant'
+        MicrosoftAppTenantId: process.env.BOT_TENANT_ID
```
Comment (corinagum) [inline js/samples/04.ai-apps/a.teamsChefBot/src/index.ts:38]: If the setup is multitenant, wouldn't this line be empty?

**Review by corinagum — APPROVED** [review]

**Comment by Benjiiim** [comment]
It would be interesting to know what make you change your mind on this one @corinagum :-)

**Pull request merged**